### DISCUSSION TITLE
I'm unable to connect to an external MariaDB database using its IP address. When I run the 'prisma db pull' command, I receive an 'Error: P1001'. However, all access permissions for the database are granted and there's no mistake on my end. Moreover, both MySQL Workbench and my Node.js server are able to establish a connection successfully, but only Prisma is not working. Can you explain why this is happening?

### DISCUSSION THREAD

**Discussion by won98** [prisma/prisma#21490] [Q&A]
### Bug description

I'm unable to connect to an external MariaDB database using its IP address. When I run the 'prisma db pull' command, I receive an 'Error: P1001'. However, all access permissions for the database are granted and there's no mistake on my end. Moreover, both MySQL Workbench and my Node.js server are able to establish a connection successfully, but only Prisma is not working. Can you explain why this is happening?

### How to reproduce

DATABASE_URL= mysql://root@localhost:3306/1111?sslaccept=strict

### Expected behavior

Error: P1001

Can't reach database server at `101.101.210.78`:`3306`

Please make sure your database server is running at `101.101.210.78`:`3306`.

### Prisma information

"prisma": "^5.4.2",

### Environment & setup

windeow, nest.js

### Prisma Version

"prisma": "^5.4.2",

**Comment by won98** [comment]
my schema.prisma file
```
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}
```

**Comment by faithfulojebiyi** [comment]
> ### Bug description
> I'm unable to connect to an external MariaDB database using its IP address. When I run the 'prisma db pull' command, I receive an 'Error: P1001'. However, all access permissions for the database are granted and there's no mistake on my end. Moreover, both MySQL Workbench and my Node.js server are able to establish a connection successfully, but only Prisma is not working. Can you explain why this is happening?
> 
> ### How to reproduce
> DATABASE_URL= mysql://root@localhost:3306/1111?sslaccept=strict
> 
> ### Expected behavior
> Error: P1001
> 
> Can't reach database server at `101.101.210.78`:`3306`
> 
> Please make sure your database server is running at `101.101.210.78`:`3306`.
> 
> ### Prisma information
> "prisma": "^5.4.2",
> 
> ### Environment & setup
> windeow, nest.js
> 
> ### Prisma Version
> "prisma": "^5.4.2",

Same issue

**Comment by ia-jortega** [comment]
Hello, I am also having this issue but the difference is I am using "Bun" JS runtime. 

Architecture:
freshly comissioned WSL2 Ubuntu latest version
Installed bun
bun add prisma 
bunx prisma init --datasource-provider mysql (Generated same schema as OP)
bunx prisma migrate dev
(DATABASE_URL="mysql://root:password@localhost:3306/world")


I have tried both MariaDB and MySql latest versions and they both return 

`Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "world" at "localhost:3306"

Error: P1001: Can't reach database server at `localhost`:`3306`

Please make sure your database server is running at `localhost`:`3306`.`

Yet I can connect to the DB servers from localhost using 3rd party clients like HeidiSQL and MySql Workbench with the same credentials. I also tried substituting localhost for 127.0.0.1 but receive same error.

**Comment by ludralph** [comment]
Hi @won98 👋 

The error message suggests checking if the database server is running at the specified IP address and port.
One potential issue could be related to the `DATABASE_URL` you provided. It appears to be using localhost in the connection string, which should be used for local connections only. Since you are trying to connect to an external database using an IP address, you should replace localhost with the actual IP address of your MariaDB server.
Here's the part of the connection string that might need to be updated (replace 101.101.210.78 with your actual database IP address):

DATABASE_URL= mysql://root@101.101.210.78:3306/1111?sslaccept=strict

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
